scipy/sparse/data.py

```diff
@@ -44,6 +44,9 @@ def _imag(self):
         return self._with_data(self.data.imag)
 
     def __neg__(self):
+        if self.dtype.kind == 'b':
+            raise NotImplementedError('inverting a sparse boolean '
+                                      'matrix is not supported')
```

Comment:
This is deprecated IINM (see @perimosocordiae 's earlier comment). With regards to your other point, the wording is a little tricky, since it is not "additive inversion" in the way we would imagine with integers.  That's why I chose just "inversion" in the error message.